cluster/gce/config-default.sh

```diff
@@ -197,8 +197,8 @@ fi
 ENCRYPTION_PROVIDER_CONFIG="${ENCRYPTION_PROVIDER_CONFIG:-}"
 if [[ -z "${ENCRYPTION_PROVIDER_CONFIG}" ]]; then
     ENCRYPTION_PROVIDER_CONFIG=$(cat << EOM | base64 | tr -d '\r\n'
-kind: EncryptionConfig
-apiVersion: v1
+kind: EncryptionConfiguration
+apiVersion: apiserver.config.k8s.io/v1
```

Comment:
This PR makes the old configs unusable. If we have to be backward compatible and you know how to handle such cases, please, share this knowledge with me or, at least, point me to the right direction. Thanks!

Reply:
Definitely needs a release note.

We could try to parse out the kind and apiVersion beforehand, then conditionally use the old parsing logic. Don't know how much work that is.

Reply:
We can register the types under the legacy core group in the scheme used for decoding. Why is that no option?

Reply:
yes, same types, multiple groups. Technically possible.

Good question about deprecation. Formally we will probably have to stick with this forever.